Framework/Algorithms/src/CalculateAsymmetry.cpp

```diff
@@ -0,0 +1,240 @@
+    // inital estimate of N0
+    const double estNormConst = estimateNormalisationConst(
+        inputWS->histogram(specNum), numGoodFrames, startX, endX);
+    // Calculate the normalised counts
+    outputWS->setHistogram(
+        specNum, normaliseCounts(inputWS->histogram(specNum), numGoodFrames));
+    // get the normalisation constant
+    const double normConst =
+        getNormConstant(outputWS, spectra[i], estNormConst, startX, endX);
+    // calculate the asymmetry
+    outputWS->mutableY(specNum) /= normConst;
+    outputWS->mutableY(specNum) -= 1.0;
+    outputWS->mutableE(specNum) /= normConst;
+
+    prog.report();
+    PARALLEL_END_INTERUPT_REGION
+  }
+  PARALLEL_CHECK_INTERUPT_REGION
+
+  // Update Y axis units
+  outputWS->setYUnit("Asymmetry");
+
+  setProperty("OutputWorkspace", outputWS);
+}
+
+/**
+ * Calculate normalisation constant after the exponential decay has been removed
+ * to a linear fitting function
+ * @param ws ::  workspace
+ * @param wsIndex :: workspace index
+ * @param estNormConstant :: estimate of normalisation constant
+ * @param startX :: the smallest x value for the fit
+ * @param endX :: the largest x value for the fit
+ * @return normalisation constant
+*/
+double CalculateAsymmetry::getNormConstant(API::MatrixWorkspace_sptr ws,
+                                           int wsIndex,
+                                           const double estNormConstant,
+                                           const double startX,
+                                           const double endX) {
+  double retVal = 1.0;
+
+  API::IAlgorithm_sptr fit;
+  fit = createChildAlgorithm("Fit", -1, -1, true);
+  std::string tmpString = getProperty("FittingFunction");
+  if (tmpString == "") {
+    g_log.warning(
+        "There is no additional function defined. Using original estimate");
+    return estNormConstant;
+  }
+  std::stringstream ss;
+
+  ss << "composite=ProductFunction;name=FlatBackground,A0=" << estNormConstant;
+  ss << ";(";
+
+  ss << "name=FlatBackground,A0=1.0"
+     << ",ties=(A0=1.0);";
+
+  ss << tmpString;
+  ss << ")";
+  std::string function = ss.str();
+  fit->setPropertyValue("Function", function);
+  fit->setProperty("InputWorkspace", ws);
+  fit->setProperty("WorkspaceIndex", wsIndex);
+  fit->setPropertyValue("Minimizer", "Levenberg-MarquardtMD");
+  fit->setProperty("StartX", startX);
+  fit->setProperty("EndX", endX);
+  fit->execute();
+
+  std::string fitStatus = fit->getProperty("OutputStatus");
+  API::IFunction_sptr result = fit->getProperty("Function");
+  std::vector<std::string> paramnames = result->getParameterNames();
+  // Check order of names
+  if (paramnames[0].compare("f0.A0") != 0) {
+    g_log.error() << "Parameter 0 should be f0.A0, but is " << paramnames[0]
+                  << '\n';
+    throw std::invalid_argument(
+        "Parameters are out of order @ 0, should be f0.A0");
+  }
+
+  if (!fitStatus.compare("success")) {
```

Comment:
The ! is correct in that line of code because of how compare works. However, I agree that it is confusing so I have changed it to 

`fitStatus=="success" `

to be explicit.